rest/model/src/main/java/io/syndesis/model/connection/CustomConnector.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.model.connection;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import io.syndesis.model.WithId;
+import io.syndesis.model.WithName;
+import io.syndesis.model.WithProperties;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(builder = CustomConnector.Builder.class)
+@SuppressWarnings("immutables")
+public interface CustomConnector extends WithId<CustomConnector>, WithName, WithProperties {
```

Comment:
Just a question for my understanding: The "CustomConnector" is specific for a custom _API_ connector or is suitable for _any_ custom connector, which is supposed to be an uploadable artefact (not scheduled yet for GA, but discussed already in Ignite GA Backlog). This was supposed to be the second extension hook beside tech extensions.

Reply:
What is the difference to `Connector` within the domain ?

Reply:
Yeah naming is hard. We had discussed this [before](https://github.com/syndesisio/syndesis-rest/pull/757#discussion_r150194277). I'm happy to rename it to something more appropriate.

The issue is that you don't give the REST API a `Connector` to create, but only parts of the data needed (`CustomConnector`), there are bits that don't really interest the end user, think Maven coordinates, or properties that can be determined from a given specification in the case of API connector.

Perhaps a name like `ConnectorSettings` or similar might be better?

Come to think of it, perhaps that `WithId` is not needed here.

Reply:
`ConnectorSettings` sounds also legit to me. I really like to avoid `CustomConnector` as it implies to be a sort of `Connector` (which is not).